src/hypothesis/core.py

```diff
@@ -945,6 +967,24 @@ def wrapped_test(*arguments, **kwargs):
             )
             arguments, kwargs, test_runner, search_strategy = processed_args
 
+            runner = getattr(search_strategy, 'runner', None)
```

Comment:
You might want to toss all of this in a function? Especially since it's error-handling, it could be nice to save a reader from it if they just want to follow the main logic.